drivers/usb/gadget/xaptum/f_psock_gadget.c

```diff
@@ -151,39 +177,52 @@ static struct usb_endpoint_descriptor hs_psock_sink_desc = {
 
 static struct usb_descriptor_header *hs_psock_descs[] = {
         (struct usb_descriptor_header *) &psock_intf,
-        (struct usb_descriptor_header *) &hs_psock_source_desc,
-        (struct usb_descriptor_header *) &hs_psock_sink_desc,
+        (struct usb_descriptor_header *) &f_psock_hs_bulk_source_desc,
+        (struct usb_descriptor_header *) &f_psock_hs_bulk_sink_desc,
+        (struct usb_descriptor_header *) &f_psock_hs_ctrl_source_desc,
+        (struct usb_descriptor_header *) &f_psock_hs_ctrl_sink_desc,
+
         NULL,
 };
 
 /**
  * Superspeed descriptors
  */
-static struct usb_endpoint_descriptor ss_psock_source_desc = {
+static struct usb_endpoint_descriptor f_psock_ss_bulk_source_desc = {
         .bLength =              USB_DT_ENDPOINT_SIZE,
         .bDescriptorType =      USB_DT_ENDPOINT,
 
         .bmAttributes =         USB_ENDPOINT_XFER_BULK,
         .wMaxPacketSize =       cpu_to_le16(1024),
 };
 
-static struct usb_ss_ep_comp_descriptor ss_psock_source_comp_desc = {
+static struct usb_ss_ep_comp_descriptor f_psock_ss_bulk_source_comp_desc = {
         .bLength =              USB_DT_SS_EP_COMP_SIZE,
         .bDescriptorType =      USB_DT_SS_ENDPOINT_COMP,
         .bMaxBurst =            0,
         .bmAttributes =         0,
         .wBytesPerInterval =    0,
 };
 
-static struct usb_endpoint_descriptor ss_psock_sink_desc = {
+static struct usb_ss_ep_comp_descriptor f_psock_ss_ctrl_comp_desc = {
+	.bLength =		sizeof f_psock_ss_ctrl_comp_desc,
+	.bDescriptorType =	USB_DT_SS_ENDPOINT_COMP,
+
+	/* the following 3 values can be tweaked if necessary */
+	/* .bMaxBurst =		0, */
+	/* .bmAttributes =	0, */
+	.wBytesPerInterval =	cpu_to_le16(16),
```

Comment:
Why 16?

Reply:
It was copied form ECM. It should be `MAX_INT_PACKET_SIZE` (64). This value tells USB how many bytes to expect to transfer per service interval (a metric only SS interrupt transfers care about).

Reply:
Leaving open for conversation but `MAX_INT_PACKET_SIZE` is being used here.